pennylane/ops/op_math/prod.py

```diff
@@ -114,6 +114,11 @@ def prod(*ops, id=None, lazy=True):
 
         @wraps(fn)
         def wrapper(*args, **kwargs):
+
+            # dequeue operators passed as arguments to the quantum function
+            leaves, _ = qml.pytrees.flatten((args, kwargs), lambda obj: isinstance(obj, Operator))
+            _ = [qml.QueuingManager.remove(l) for l in leaves if isinstance(l, Operator)]
```

Comment:
```suggestion
            for l in leaves:
                if isinstance(l, Operator):
                    qml.QueuingManager.remove(l)
```
Is this assignment merely for execute the `remove`? If so would it be ok to just for loop it